device/esp_tinyusb/tinyusb_msc.c

```diff
@@ -291,39 +291,39 @@ esp_err_t msc_storage_mount(void)
         goto fail;
     }
 
+    // Registering the FATFS object was done successfully; change the mount point.
+    // All subsequent errors depend on the filesystem.
+    s_storage_handle->mount_point = TINYUSB_MSC_STORAGE_MOUNT_APP;
+
     ret = vfs_fat_mount(drv, fs, true);
     if (ret == ESP_ERR_NOT_FOUND) {
         // If mount failed, try to format the drive
         if (s_storage_handle->fat_fs.do_not_format) {
-            ESP_LOGE(TAG, "Mount failed and do not format is set");
+            ESP_LOGD(TAG, "Mount failed and do not format is set");
             tinyusb_event_cb(TINYUSB_MSC_EVENT_FORMAT_REQUIRED);
-            goto fail;
+            ret = ESP_OK;
+            goto exit;
         }
-        ESP_LOGW(TAG, "Mount failed, trying to format the drive");
+        ESP_LOGW(TAG, "Filesystem not found; creating a new one");
         BYTE format_flags = s_storage_handle->fat_fs.format_flags;
-        ret = vfs_fat_format(format_flags);
-        if (ret != ESP_OK) {
-            ESP_LOGE(TAG, "Failed to format the drive, %s", esp_err_to_name(ret));
-            goto fail;
-        }
+        ESP_GOTO_ON_ERROR(vfs_fat_format(format_flags), fail, TAG, "Failed to format the drive");
         ESP_GOTO_ON_ERROR(vfs_fat_mount(drv, fs, false), fail, TAG, "Failed to mount FAT filesystem");
-        ESP_LOGW(TAG, "Format completed, FAT mounted successfully");
+        ESP_LOGD(TAG, "Format completed, FAT mounted successfully");
     } else if (ret != ESP_OK) {
         ESP_LOGE(TAG, "Failed to mount drive, %s", esp_err_to_name(ret));
-        goto fail; // If mount was unsuccessful, return the error
+        goto fail;
```

Comment:
Could we simply call `tinyusb_event_cb(TINYUSB_MSC_EVENT_FORMAT_FAILED);` here, instead of at the `fail` label?

Reply:
Fair enough, I missed some of the cases